### ISSUE TITLE
Allow execution of SQL SELECT queries on read-only databases

### ISSUE THREAD

**Issue by cplaetzinger** [hasura/graphql-engine#9449]
We use hasura on a read-only database. Thus, we configured a separate metadata database which is writable. Everything works fine and we can execute graphql queries and use the web console to browse the data. However, we can not execute any query using the SQL feature of the console. 

A simple `SELECT * FROM products` leads to the following error:
![image](https://user-images.githubusercontent.com/1055217/220089415-e466dd28-e29a-42aa-b729-ccac72f7ca48.png)

Hasura is started with the follwing ENV variables set:
```
HASURA_GRAPHQL_ADMIN_SECRET=...
HASURA_GRAPHQL_DATABASE_URL=postgres://...
HASURA_GRAPHQL_DEV_MODE=false
HASURA_GRAPHQL_ENABLE_CONSOLE=true
HASURA_GRAPHQL_ENABLED_LOG_TYPES=startup, http-log, webhook-log, websocket-log, query-log
HASURA_GRAPHQL_LOG_LEVEL=info
HASURA_GRAPHQL_METADATA_DATABASE_URL=postgres://...
```
During my research I've found #7029 which seems to describe an similar issue. However we are using v2.6.1.

**Comment by Varun-Choudhary** [comment]
Now we added the console support for read_only mode to run SELECT query in read_only. Closing this issue.

**Comment by cplaetzinger** [comment]
Thank you for the fast response. Looking forward to try it out. Which version will include the fix?

**Comment by Varun-Choudhary** [comment]
Hey @cantaspinar, It is available from `v2.20.0-beta.1`
